src/compiler/checker.ts

```diff
@@ -43367,12 +43366,12 @@ namespace ts {
             return false;
         }
 
-        function isRequiredInitializedParameter(parameter: ParameterDeclaration | JSDocParameterTag): boolean {
-            return !!strictNullChecks &&
+        function isRequiredInitializedParameter(parameter: ParameterDeclaration | JSDocParameterTag, checkSyntacticModifier?: boolean, ignoreStrictNullChecks?: boolean): boolean {
+            return (ignoreStrictNullChecks || !!strictNullChecks) &&
                 !isOptionalParameter(parameter) &&
                 !isJSDocParameterTag(parameter) &&
                 !!parameter.initializer &&
-                !hasSyntacticModifier(parameter, ModifierFlags.ParameterPropertyModifier);
+                (!checkSyntacticModifier || !hasSyntacticModifier(parameter, ModifierFlags.ParameterPropertyModifier));

```

Comment:
I read this as: when checkSyntacticModifier=false, we are supposed to treat all parameters with initialisers as required, even if they are parameter properties. (normally parameter properties with initialisers are optional.)

Why? I kind of understand the original rule: parameter properties with initialisers don't need an argument, although I don't understand why they're special. But I don't understand why they now need an exemption. I guess it's so that the declaration emitter can distinguish between the emitted parameter (needs to add `| undefined`) and the property declaration (needs no `undefined`).